checkups/kubevirt-vm-latency/vmlatency/internal/client/client.go

```
@@ -32,6 +38,20 @@ func New() (*Client, error) {
 	return &Client{c}, nil
 }
 
-func (c *Client) UpdateConfigMap(_, _ string, _ map[string]string) error {
+func (c *Client) UpdateConfigMap(namespace, name string, date map[string]string) error {
+	const updateConfigMapTimeout = time.Minute * 3
+	ctx, cancel := context.WithTimeout(context.Background(), updateConfigMapTimeout)
+	defer cancel()
```

Please consider removing it for now, and adding it in the future if there will be a need.
I suggest for now to use `context.Background` for simplicity.

---

It would be helpful if you explain why.

@ormergi , from your side, I am interesting also to understand why this is needed.
Are you saying that any of these calls are blocking? It does not seem so to me, but I am not really sure.

---

All our calls to the K8s API were called with the `context.Background`.
We haven't considered a policy for timeouts yet.
Why is this case different?

---

1. From what I saw in the code of the client both operations are blocking, in general it better to limit such operations
    with a timeout and fail faster.

2. We do set a timeouts at the framework [[1]](https://github.com/kiagnose/kiagnose/blob/main/kiagnose/internal/checkup/checkup.go#L290) [[2]](https://github.com/kiagnose/kiagnose/blob/main/kiagnose/internal/checkup/checkup.go#L309) (there is a tech death to use context and timeouts to allow graceful shutdown)

3. IMO there is no need to come up with formal timeouts policy, time framing operations against an API endpoint
is pretty standard as far as I know.

---

> 1. From what I saw in the code of the client both operations are blocking, in general it better to limit such operations
>    with a timeout and fail faster.

The operations are indeed blocking.

> 2. We do set a timeouts at the framework [[1]](https://github.com/kiagnose/kiagnose/blob/main/kiagnose/internal/checkup/checkup.go#L290) [[2]](https://github.com/kiagnose/kiagnose/blob/main/kiagnose/internal/checkup/checkup.go#L309) (there is a tech death to use context and timeouts to allow graceful shutdown)

There is no timeout on low level K8s API calls in the kiagnose core.

> 3. IMO there is no need to come up with formal timeouts policy, time framing operations against an API endpoint
>    is pretty standard as far as I know.

I agree with the need, but suggest to add it in a later stage.

---

I prefer to do it now and not drag more death for later, as you can see its pretty straight forward (passing context with timeout down the chain..)

---

The way I see it is introducing a new thing to the project, same as the error type that was discussed yesterday in PR #41.

---

I have seen no usage of timeouts to API calls that are expected to perform a simple operation. The only usage I have seen this used was for `wait*` operations, which by definition are long blocking (waiting for a condition to be reached).

I think we should avoid adding new patterns implicitly. Please try to suggest such improvements through dedicated PR/s.
The basic guideline is what was merged so far and if there is a problem that requires changing the existing, a change suggesting that should be placed, fixing it.

---

Thanks for explaining, I will remove it.

Regarding adding new patterns implicitly, I think its a bit excessive when you need to introduce some new functionality
when there is no similar example for it in the code base.